zer0-os/zOS
# feat: (matrix) edit message

### What does this do? 
Introduces the capability to handle and process edited messages. Key changes include:
- Implementing functionality to determine if a message is an edit event.
- Refactoring message handling logic to account for edited messages and update the original message content accordingly.
- Introducing mechanisms to listen for edited message events and respond with the appropriate updates.
- Refactoring related functions for better clarity and reduced code repetition.

### Why are we making this change?
To ensure that our matrix client can effectively handle message edits.

### How do I test this?
Open the messenger > Send a message > Click the three dots message menu > Select `Edit` > Edit the message and submit > Check message has been edited.

### Key decisions and Risk Assessment:
  #### Things to consider:
  1. How will this affect security?
  1. How will this affect performance?
  1. Does this change any APIs?


NOTE: The flicker/movement of the edited message here will be resolved in a follow up PR.

https://github.com/zer0-os/zOS/assets/39112648/a7dfe201-32a4-4cae-a469-7f939ac1e853
